workbench-core/base/src/services/metadataService.test.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { MetadataService } from './metadataService';
+
+describe('metadata service', () => {
+  const metadataService = new MetadataService();
+  interface EntityA {
+    id: string;
+    sk: string;
+    pk: string;
+    productName: string;
+  }
+  interface EntityB {
```

Comment:
you can use interface extension to reduce duplicated properties by adding `interface ____ extends EntityA` and then just adding whatever properties you need to.